scripts/spcgeonode/django/initialize.py

```diff
@@ -101,16 +101,22 @@
 )
 app.skip_authorization = True
 _host = os.getenv('HTTPS_HOST', "") if os.getenv('HTTPS_HOST', "") != "" else os.getenv('HTTP_HOST')
-_port = os.getenv('HTTPS_PORT') if os.getenv('HTTPS_HOST', "") != "" else os.getenv('HTTP_PORT')
-if _port and _port not in ("80", "443"):
+_port = os.getenv('HTTPS_PORT') if os.getenv('HTTPS_HOST', "") != "" else os.getenv('HTTP_PORT', '80')
+# default port is 80
+_protocols = {
+    "80": "http://",
+    "443": "https://"
+}
+if _port not in _protocols:
     redirect_uris = [
         'http://{}:{}/geoserver'.format(_host, _port),
         'http://{}:{}/geoserver/index.html'.format(_host, _port),
     ]
 else:
+    # Make sure protocol string match with GeoServer Redirect URL's protocol string
     redirect_uris = [
-        'http://{}/geoserver'.format(_host),
-        'http://{}/geoserver/index.html'.format(_host),
+        '{}{}/geoserver'.format(_protocols[_port], _host),
+        '{}{}/geoserver/index.html'.format(_protocols[_port], _host),
```

Comment:
Make sure protocol strings match. If it doesn't match, OAuth redirect will not work.
It can be fixed in the admin page, but container restart will revert the changes to default again.